platform/fdo/cog-platform-fdo.c

```diff
@@ -1981,6 +1998,24 @@ clear_wayland (void)
     wl_display_disconnect (wl_data.display);
 }
 
+static void *
+check_supported(void *data G_GNUC_UNUSED)
+{
+    if (init_wayland(NULL)) {
```

Comment:
Same here, `init_wayland` does protocol roundtrips and loads cursor data. That's not viable for a simple check as intended here.

Reply:
Indeed, I have created #331 to take care of this.